samples/sample_common/src/sample_utils.cpp

```diff
@@ -858,6 +858,58 @@ void CSmplYUVWriter::Close()
     m_bInited = false;
 }
 
+mfxStatus GetChromaSize(const mfxFrameInfo & pInfo, mfxU32 & ChromaW, mfxU32 & ChromaH)
+{
+    mfxStatus sts = MFX_ERR_NONE;
+
+    switch (pInfo.FourCC)
+    {
+    case MFX_FOURCC_YV12:
+    {
+        ChromaW = (pInfo.CropW + 1) / 2;
+        ChromaH = (pInfo.CropH + 1) / 2;
+        break;
+    }
+    case MFX_FOURCC_NV12:
+    {
+        ChromaW = pInfo.CropW;
+        ChromaH = (pInfo.CropH + 1) / 2;
+        break;
+    }
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P210:
+    {
+        ChromaW = pInfo.CropW;
+        ChromaH = pInfo.FourCC == MFX_FOURCC_P210 ? (mfxU32)pInfo.CropH : (mfxU32)(pInfo.CropH + 1) / 2;
+        break;
+    }
+
+    case MFX_FOURCC_RGB4:
+    case 100: //DXGI_FORMAT_AYUV
```

Comment:
why dxgi enum can be here?

Reply:
In fact, I don't know why either. I just refer to the implementation WriteNextFrame(). If necessary, I can remove it. Thanks.